src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java

```diff
@@ -78,6 +79,16 @@ private static void sendOutOfBandTelemetryMessage(
     String stackTrace = maskStacktrace(sw.toString());
     value.put("Stacktrace", stackTrace);
     value.put("Exception", ex.getClass().getSimpleName());
+    // For SQLFeatureNotSupportedExceptions, add in reason for failure as "<function name> not
+    // supported"
+    if (value.get("SQLState").toString().contains(SqlState.FEATURE_NOT_SUPPORTED)) {
+      String reason = "";
+      StackTraceElement[] stackTraceArray = ex.getStackTrace();
+      if (stackTraceArray.length >= 1) {
+        reason = ex.getStackTrace()[0].getMethodName() + " not supported";
```

Comment:
Just one question here. Is the first element of ex.getStackTrace() the one we want? I remember in some similar cases the first one is not what we want. So just want to double check.

Reply:
Good point! In some setups, it's the 0th element and in some, it's the 1st. With our driver version, it is 0. I ran through the debugger with multiple unsupported functions and this gives me the behavior I want, which is "<function name> not supported".